frontend/src/components/ViewActivities/editActivity.js

```diff
@@ -22,7 +22,16 @@ class EditActivity extends React.Component {
   constructor(props){
     super(props);
 
-    this.state = {startTzChanged: false, endTzChanged: false};
+    const activity = props.activity;
+    this.state = {
+      startTzChanged: false, 
+      endTzChanged: false, 
+      startTimeDateChanged: false,
+      endDateFiller: time.getISODate(getField(activity, DB.ACTIVITIES_END_TIME), 
+        getField(activity, DB.ACTIVITIES_END_TZ)),
+      endTimeFiller: time.get24hTime(getField(activity, DB.ACTIVITIES_END_TIME), 
+        getField(activity, DB.ACTIVITIES_END_TZ))
+    };
```

Comment:
I would diff this PR off of #124 so that it does not look like these are all new.

Reply:
These ARE new in this PR.